pkg/kp/dsstore/consul_store.go

```diff
@@ -214,6 +214,31 @@ func (s *consulStore) MutateDS(
 	return ds, nil
 }
 
+func (s *consulStore) Disable(id fields.ID) (fields.DaemonSet, error) {
+	s.logger.Infof("Attempting to disable '%s' in store now", id)
+
+	mutator := func(dsToUpdate fields.DaemonSet) (fields.DaemonSet, error) {
+		dsToUpdate.Disabled = true
+		return dsToUpdate, nil
+	}
+	newDS, err := s.MutateDS(id, mutator)
+
+	// Delete the daemon set because there was an error during mutation
```

Comment:
Is Delete() the safest way for this function to fail? I understand the following:

`Disable` has the semantic: "Don't schedule this pod onto any more nodes"
`Delete` has the semantic: "Remove this DS from consul and _unschedule all of its pods_"

If my characterization is correct, it's not at all safe to Delete the DaemonSet. This function should probably do nothing in case of Mutation error.